hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/AppSchedulingInfo.java

```diff
@@ -129,8 +132,32 @@ public AppSchedulingInfo(ApplicationAttemptId appAttemptId, String user,
     updateContext = new ContainerUpdateContext(this);
     readLock = lock.readLock();
     writeLock = lock.writeLock();
+
+    this.defaultResourceRequestAppPlacementType =
+        getDefaultResourceRequestAppPlacementType();
```

Comment:
Configuration has this getClass() method, should we use this?

Reply:
here, actual value is coming from env variable
      String appPlacementClass = applicationSchedulingEnvs.get(
          ApplicationSchedulingConfig.ENV_APPLICATION_PLACEMENT_TYPE_CLASS);

This is just a fallback in case the env variable is not set.